src/database.ts

```diff
@@ -1348,12 +1349,28 @@ class Database extends GrpcServiceObject {
       delete gaxOpts.pageToken;
     }
 
-    return this.requestStream({
-      client: 'SpannerClient',
-      method: 'listSessionsStream',
-      reqOpts,
-      gaxOpts,
-    });
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this;
+    const transform = function (
+      this: Transform,
+      chunk: databaseAdmin.spanner.v1.ISession,
+      enc: string,
+      callback: Function
+    ) {
+      const session = self.session(chunk.name!);
+      session.metadata = chunk;
+      callback(null, session);
+    };
+
+    return new pumpify.obj([
```

Comment:
Sorry for the possibly stupid question here, but do we need to introduce `pumpify` for this? Would it not be just as easy (and one dependency less) to do something like the following:

```typescript
    const req = this.requestStream({
      client: 'SpannerClient',
      method: 'listSessionsStream',
      reqOpts,
      gaxOpts,
    })
      .on('error', e => req.destroy(e))
      .pipe(new Transform({objectMode: true, transform}));
    return req;
```

Reply:
`pumpify` is used throughout the different clients as well ([bigtable](https://github.com/googleapis/nodejs-bigtable/blob/6d79c48a07111f6def752b636def22d17f748956/src/table.ts#L813), [storage](https://github.com/googleapis/nodejs-storage/blob/621e6f4723d48511c94f11cf0aa7d2bc20516b65/src/file.ts#L1738), [storage](https://github.com/googleapis/nodejs-storage/blob/621e6f4723d48511c94f11cf0aa7d2bc20516b65/src/file.ts#L1304)).

In case of using `pipe()` as `source.pipe(dest)` the source stream will not be destroyed if dest stream breaks, emits close,  or an error. `pumpify` ensures that source stream is destroyed.

Reply:
Sounds reasonable.